### PR TITLE
Add custom user recover password view

### PR DESCRIPTION
**Pull request by norkans7**

### CODE REVIEW

File: temba/urls.py [hunk 1]
```diff
@@ -36,6 +37,7 @@
     url(r"^", include("temba.triggers.urls")),
     url(r"^relayers/relayer/sync/(\d+)/$", sync, {}, "sync"),
     url(r"^relayers/relayer/register/$", register, {}, "register"),
+    url(r"users/user/forget/", RedirectView.as_view(pattern_name="orgs.user_forget", permanent=True)),
```
Comment (norkans7) [inline temba/urls.py:40]: Make sure we redirect the smartmin forget view to the one we have in RapidPro

File: temba/orgs/views.py [hunk 2]
```diff
@@ -678,6 +680,46 @@ def form_valid(self, form):
...
+
+            if user:
+                subject = _("Password Recovery Request")
+                template = "orgs/email/user_forget"
+
+                token = "".join(random.choice(string.ascii_uppercase + string.digits) for x in range(32))
+                RecoveryToken.objects.create(token=token, user=user)
+                FailedLogin.objects.filter(username__iexact=user.username).delete()
+
+                context = dict(user=user, path=f'{reverse("users.user_recover", args=[token])}')
```
Comment (norkans7) [inline temba/orgs/views.py:712]: Re used the same approach we use elsewhere to send emails

**Comment by norkans7** [comment]
Replaces the other approach in https://github.com/nyaruka/rapidpro/pull/3335 and https://github.com/nyaruka/smartmin/pull/148

**Comment by rowanseymour** [comment]
Screenshots please

**Comment by norkans7** [comment]
RapidPro - Visually build nationally scalable mobile applications 2021-07-05 18-34-29

**Review by rowanseymour — APPROVED** [review]

File: temba/urls.py (hunk 1 above)
Comment (rowanseymour) [reply temba/urls.py:40]: Do we need this? Is there anywhere we link to this?

Comment (norkans7) [reply temba/urls.py:40]: We do not but one can type the URL and get to it so I wanted to make sure we still go to the view we want which as a side effect will also catch the missed link if any

**Pull request merged**